test/end2end_test.go

```diff
@@ -697,12 +733,34 @@ func (te *test) startServerWithConnControl(ts testpb.TestServiceServer) *listene
 	return l.(*listenerWrapper)
 }
 
-// startServer starts a gRPC server listening. Callers should defer a
-// call to te.tearDown to clean up.
+// startServer starts a gRPC server exposing the provided TestService
+// implementation. Callers should defer a call to te.tearDown to clean up
 func (te *test) startServer(ts testpb.TestServiceServer) {
 	te.listenAndServe(ts, net.Listen)
 }
 
+// startServers starts 'num' gRPC servers exposing the provided TestService.
+func (te *test) startServers(ts testpb.TestServiceServer, num int) {
+	for i := 0; i < num; i++ {
+		te.startServer(ts)
+		te.srvs = append(te.srvs, te.srv.(*grpc.Server))
+		te.hSrvs = append(te.hSrvs, te.hSrv)
+		te.srvAddrs = append(te.srvAddrs, te.srvAddr)
+		te.srv = nil
+		te.hSrv = nil
+		te.srvAddr = ""
+	}
+}
+
+// setHealthServingStatus is a helper function to set the health status.
+func (te *test) setHealthServingStatus(service string, status healthpb.HealthCheckResponse_ServingStatus) {
+	if te.hSrv != nil {
```

Comment:
I guess the main purpose of this function is the nil check.
If so, I think it should be OK to call it directly in the tests without checking nil. Because if `hSrv` is nil, the test will fail, and we will know. It won't be flaky. And we never have code checked in with nil pointer panic.

For the type assertion, can we make `hSrv` type `*health.Server`? If we never use a non-standard health server implementation.

Another use of this functions would be for `startServers`, where we have multiple health servers. But that's not used anywhere today.

Reply:
I removed the function for now, since it is not called by anyone (makes vet unhappy). Will revisit again if I need it once I work on the follow up PR to fix the health_check tests. Thanks.